core/document/src/layers/polyline.rs

```diff
@@ -0,0 +1,24 @@
+use super::style;
+use super::LayerData;
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct PolyLine {
+	points: Vec<kurbo::Point>,
+	style: style::PathStyle,
+}
+
+impl PolyLine {
+	pub fn new(points: Vec<impl Into<kurbo::Point>>, style: style::PathStyle) -> PolyLine {
+		PolyLine {
+			points: points.into_iter().map(|it| it.into()).collect(),
+			style,
+		}
+	}
+}
+
+impl LayerData for PolyLine {
+	fn render(&self) -> String {
+		let points = self.points.iter().map(|p| format!("{},{}", p.x, p.y)).collect::<Vec<_>>().join(" ");
```

Comment:
would it be more efficient to use a fold here? To avoid creating a vec that is  destroyed

Reply:
Want to submit a quick followup PR and I can rubber stamp it?

Reply:
Yeah, but I think it do not matter a lot , since this operation should be just one off for layer, right ?

Reply:
the render function is called every "frame" so using some sort of caching would be even better, this could otherwise actually become a performance bottleneck